### PR TITLE
Document how to publish a new module version for a branch based module.

### PR DESCRIPTION
**Pull request by aaabdelgany**
### What
Document how to publish a new module version for a branch based module.

docs/cloud-docs/api-docs/private-registry/modules.mdx
### Why
This is a workflow that users are able to take advantage of via the ui and api and the documentation should be updated to reflect that.
### Screenshots

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. HCP Terraform ).

#### Content
- [x] API documentation and the API Changelog have been updated. 

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

**Review by paulanunda — APPROVED** [review]

**Review by BrianMMcClain — CHANGES_REQUESTED** [review]
Thanks! Just had one quick question to clarify some different uses of the `commit_sha` parameter that I've seen.

File: website/docs/cloud-docs/api-docs/changelog.mdx [hunk 1]
```diff
@@ -8,6 +8,8 @@ description: >-
 # API Changelog
 
 Keep track of changes to the API for HCP Terraform and Terraform Enterprise.
+## 2024-05-23
+* Update [Registry Modules API](/terraform/cloud-docs/api-docs/private-registry/modules#create-a-module-version) to detail how to publish a new version for a branch based module.
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/changelog.mdx:12]: Style suggestions (we only capitalize product names)
```suggestion
* Update the [registry modules API](/terraform/cloud-docs/api-docs/private-registry/modules#create-a-module-version) for publishing new versions of branch based modules.
```

File: website/docs/cloud-docs/api-docs/changelog.mdx [hunk 2]
```diff
@@ -8,6 +8,8 @@ description: >-
 # API Changelog
 
 Keep track of changes to the API for HCP Terraform and Terraform Enterprise.
+## 2024-05-23
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/changelog.mdx:11]:
```suggestion

## 2024-05-23
```

**Review by rkoron007 — APPROVED** [review]
Added suggestions for style but looks good!

File: website/docs/cloud-docs/api-docs/private-registry/modules.mdx [hunk 3]
```diff
@@ -535,9 +535,11 @@ This POST endpoint requires a JSON object with the following properties as a req
 Properties without a default value are required.
 
 | Key path                  | Type   | Default | Description                           |
-| ------------------------- | ------ | ------- | ------------------------------------- |
-| `data.type`               | string |         | Must be `"registry-module-versions"`. |
-| `data.attributes.version` | string |         | A valid semver version string.        |
+| ---------------------------- | ------ | ------- | ---------------------------------------------------- |
+| `data.type`                  | string |         | Must be `"registry-module-versions"`.                |
+| `data.attributes.version`    | string |         | A valid semver version string.                       |
+| `data.attributes.commit_sha` | string |         | A valid sha for the module version to ingress        |
```
[the remark this answers is not in the document]
Comment (aaabdelgany) [reply website/docs/cloud-docs/api-docs/private-registry/modules.mdx]: great catch, just addressed it. ty!

File: website/docs/cloud-docs/api-docs/private-registry/modules.mdx [hunk 4]
```diff
@@ -546,7 +548,8 @@ Properties without a default value are required.
   "data": {
     "type": "registry-module-versions",
     "attributes": {
-      "version": "1.2.3"
+      "version": "1.2.3",
+      "commit_sha": "abcdef12345"
```
Comment (BrianMMcClain) [inline website/docs/cloud-docs/api-docs/private-registry/modules.mdx:552]:
```suggestion
      "commit-sha": "abcdef12345"
```

**Review by BrianMMcClain — APPROVED** [review]

File: website/docs/cloud-docs/api-docs/private-registry/modules.mdx (hunk 4 above)
Comment (aaabdelgany) [reply website/docs/cloud-docs/api-docs/private-registry/modules.mdx]: my apologies for missing this, ty brian!

**Pull request merged**